src/citationnet/static/pieChart_vega-lite.js

```diff
@@ -0,0 +1,88 @@
+const forChartDiv = document.getElementById("forChartDiv");
+const forChartDivvis = document.getElementById("forChartDivvis");
+const forChartDivloader = document.getElementById("forChartDivloader");
+
+dragElement(forChartDiv);
+
+function toggleFORChart() {
+    if (forChartDiv.hidden) {
+        showFOR();
+    } else {
+        hideFOR();
+    }
+}
+
+async function showFOR() {
+    document.getElementById("btnFORChart").style.fontWeight = "bolder";
+    forChartDivvis.hidden = true;
+    forChartDivloader.hidden = false;
+    forChartDiv.hidden = false;
+
+    await loadFOR();
+
+    forChartDivvis.hidden = false;
+    forChartDivloader.hidden = true;
+}
+
+function hideFOR() {
+    document.getElementById("btnFORChart").style.fontWeight = "normal";
+    forChartDiv.hidden = true;
+}
+
+async function loadFOR() {
+    const stats = await window.net.getStats();
+    const resp = await fetch("/static/vegaLiteSpec.json");
+    const vegaLiteSpec = await resp.json();
+    const domain = stats.map(value => value.category);
+    const range = stats.map(value => "#".concat(value.color));
+
+    vegaLiteSpec.data.values = stats;
+    vegaLiteSpec.encoding.color.scale.domain = domain;
+    vegaLiteSpec.encoding.color.scale.range = range;
+
+    console.log(vegaLiteSpec);
+
+    vegaEmbed('#forChartDivvis', vegaLiteSpec);
+}
+
+function dragElement(elmnt) {
+    var pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
+    if (document.getElementById(elmnt.id + "header")) {
+        /* if present, the header is where you move the DIV from:*/
+        document.getElementById(elmnt.id + "header").onmousedown = dragMouseDown;
+    } else {
+        /* otherwise, move the DIV from anywhere inside the DIV:*/
+        elmnt.onmousedown = dragMouseDown;
```

Comment:
This is fine, but I'd recommend using `addEventListener` instead for more control and reusability of the handler (as well as being able to detach it properly if/when needed!)

Reply:
Just thought that you may also want to consider using the same handler to add a [touch event](https://developer.mozilla.org/en-US/docs/Web/API/Touch_events/Using_Touch_Events#interfaces) listener (though I haven't tested the application with a touch screen, so I don't know to what extent it's worth supporting touch).